dlpy/transformers/bert_model.py

```diff
@@ -107,53 +105,153 @@ def __init__(self, conn, cache_dir, name='bert_base_uncased', n_classes=None,
-
-        if self._verbose:
-            print("NOTE: base BERT model loaded.")
-
-        # instantiate BERT _tokenizer
-        self._tokenizer = BertTokenizer.from_pretrained(name)
+        elif hf_base_name == 'roberta':
+            self._base_model = RobertaModel.from_pretrained(self._base_name,
+                                                            cache_dir=cache_dir,
+                                                            num_hidden_layers=num_hidden_layers)
+            self._tokenizer = RobertaTokenizer.from_pretrained(self._base_name)
+            self._config = RobertaConfig.from_pretrained(self._base_name,
+                                                         cache_dir=cache_dir,
+                                                         num_hidden_layers=num_hidden_layers).to_dict()
+        else:   # all distil* models
+            self._base_model = DistilBertModel.from_pretrained(self._base_name,
+                                                               cache_dir=cache_dir,
+                                                               n_layers=num_hidden_layers)
+            self._tokenizer = DistilBertTokenizer.from_pretrained(self._base_name)
+            self._config = DistilBertConfig.from_pretrained(self._base_name,
+                                                            cache_dir=cache_dir,
+                                                            n_layers=num_hidden_layers).to_dict()
+            
+        # set model-specific quantitites
+        self._set_bert_type_info(hf_base_name)
         
+        if self._verbose:
+            print("NOTE: base model " + self._base_name + " loaded.")
+       
         # load embedding table (token | position | segment)
-        self._load_embedding_table()
+        self._load_embedding_table(save_embedding)
             
         if self._config['max_position_embeddings'] < max_seq_len:
             raise DLPyError('The specified maximum sequence length exceeds the maximum position embedding.')
         else:
             self._max_seq_len = max_seq_len
+            
+        # set input layers
+        if self._use_segment_embedding:
+            self.ds_layers = dict(token_input=BertCommon['layer_names']['token_input'],
+                                  position_input=BertCommon['layer_names']['position_input'],
+                                  segment_input=BertCommon['layer_names']['segment_input'])
+        else:
+            self.ds_layers = dict(token_input=BertCommon['layer_names']['token_input'],
+                                  position_input=BertCommon['layer_names']['position_input'])
+        
+    def _set_bert_type_info(self, bert_variant_name):
+        if bert_variant_name in ['bert', 'camembert']:
+            self._use_pooling_layer = True
+            self._position_embedding_offset = 0
+        elif 'roberta' in bert_variant_name:
```

Comment:
Yes, although I didn't realize that I had missed this camembert reference.  I will rework this section with my updated pull request so that the only "contains" check is for distil.  I need to be able to cover distilbert and distilroberta.